app/controllers/fee_calculator_controller.rb

```diff
@@ -3,5 +3,7 @@
 class FeeCalculatorController < ApplicationController
   def index
     @message = 'Fee scheme'
+    @fee_schemes = FeeCalculator.new.fee_schemes
+    @fee_scheme = FeeCalculator.new.fee_scheme('AGFS', '2022-09-29')
```

Comment:
There are a couple of different approaches that you might consider here.

Firstly, you do not need to create two instances of the `FeeCalculator` class, so this could be;

```suggestion
    fee_calculator = FeeCalculator.new
    @fee_schemes = fee_calculator.fee_schemes
    @fee_scheme = fee_calculator.fee_scheme('AGFS', '2022-09-29')
```

The second approach is to separate this into two separate services. Ideally, a service has a single purpose, which is carried out using `Service.new(<params>).call` (which can optionally be shortened to `Service.call(<params>)`). In this case the code in the controller would look something like;

```suggestion
    @fee_schemes = FeeCalculator::FeeSchemes.new.call
    @fee_scheme = FeeCalculator::FeeScheme.new('AGFS', '2022-09-29').call
```